### ISSUE TITLE
ppltasks.cpp: Breaking change - New OLE32.dll import prevents process mitigations

### ISSUE THREAD

**Issue by fbrosseau** [microsoft/STL#3257]
Hello,

Commit https://github.com/microsoft/STL/commit/80ebe871c7af413e5f3adced95e84ec238a884a2 introduced a new import for ole32.dll in shipped msvcp140.dll. 

This is a problem for us, because this import of ole32 cascades into eventually loading user32, immediately at process startup.
This is indirectly a breaking change, because our program needs to enable process mitigations (hardening) on itself, and the system will refuse to enable anti-win32k mitigations if user32 is already loaded in the process.

Workarounds possibly exist for our specific usecase (such as not enabling the mitigation on ourselves and instead having a parent process enable them on a child at the time of CreateProcess), but it remains a breaking change - [SetProcessMitigationPolicy](https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-setprocessmitigationpolicy) is explicitly designed to set mitigations on yourself.

Could ole32 be delay-loaded only when ppltasks are used? 

Edit: the commit I linked does mention the intent was that ole32 should be delay-loaded. This does not appear to be the case in msvcp140 14.34.31931.0 (what I currently have on latest VS) and beyond?

This is also DevCom-10217462.

**Comment by sylveon** [comment]
This is weird, I'm able to enable DisallowWin32kSystemCalls even when user32.dll is loaded (my process directly links user32.dll and enables said policy), the API does not return an error. Do you have a minimum repro?

**Comment by fbrosseau** [comment]
Sure. Please ignore the messy code, it's not easy to avoid using the CRT : )

```c++
// cl.exe -nologo -Gm- -GR- -EHa- -Oi -Zi test.cpp -link -nodefaultlib -subsystem:console kernel32.lib -debug:full

#include <Windows.h>

HANDLE g_stdout;

PCWSTR strchr(PCWSTR str, WCHAR find)
{
    while (*str != find)
    {
        ++str;
    }

    return str;
}

void WriteStdout(PCWSTR str)
{
    while (*str)
    {
        DWORD written;
        WriteFile(g_stdout, str, sizeof(WCHAR), &written, nullptr);
        ++str;
    }
}

int mainCRTStartup()
{
    g_stdout = GetStdHandle(STD_OUTPUT_HANDLE);

    PCWSTR libraryToLoad = strchr(GetCommandLineW(), ' ') + 1;
    while (*libraryToLoad == ' ')
    {
        ++libraryToLoad;
    }

    HMODULE mod = LoadLibraryW(libraryToLoad);

    WriteStdout(mod ? L"Loaded ": L"Failed to load ");
    WriteStdout(libraryToLoad);
    WriteStdout(L"\n");

    PROCESS_MITIGATION_SYSTEM_CALL_DISABLE_POLICY SystemCallPolicy{};
    SystemCallPolicy.DisallowWin32kSystemCalls = TRUE;

    if (!SetProcessMitigationPolicy(ProcessSystemCallDisablePolicy,
        &SystemCallPolicy,
        sizeof(SystemCallPolicy)))
    {
        DWORD err = GetLastError();
        WriteStdout(L"FAIL\n");
        return err;
    }

    WriteStdout(L"OK\n");
    return 0;
}
```

```cmd
test.exe kernel32.dll -> ok (it's already loaded anyway)
test.exe user32.dll -> not ok
test.exe ole32.dll -> not ok (implies user32.dll)
test.exe msvcp140.dll -> not ok (implies ole32.dll)
```

**Comment by sylveon** [comment]
Ah, my bad. Looks like my code wasn't actually enabling this, even though I thought I recalled it did. Your code does repro the problem.

For reference, build by doing
```
cl /c test.cpp
link /NODEFAULTLIB /SUBSYSTEM:CONSOLE test.obj kernel32.lib
```

I don't see anywhere in the commited code where it would imply delay loading. However, I found this funny outdated comment:
![image](https://user-images.githubusercontent.com/6440374/205414999-27714713-c079-4a3b-b0a8-213cf72be609.png)
